### ISSUE TITLE
Just starting.  Need help with MPI

### ISSUE THREAD

**Issue by samcable** [AMReX-Codes/amrex#297]
I'm just getting started with AMReX.  Trying to run on a Mac, macOS High Sierra.  I am trying to get the Fortran "Hello World" to run and compile with MPI.  It looks like the mpicxx linker cannot find any piece of the MPI library.  This is not a problem when I use mpicxx to compile my own test codes.  I read through "Building with GNU Make" including the section about problems with Macs.  Nothing there seems to help.  Would be most grateful for any ideas.

**Comment by zingale** [comment]
can you post the whole error message from compilation?

**Comment by samcable** [comment]
Here's the error.  Happens during linking.


Undefined symbols for architecture x86_64:
  "_mpi_allreduce_", referenced from:
      ___amrex_parallel_module_MOD_amrex_parallel_reduce_min_rv in AMReX_parallel_mod.o
      ___amrex_parallel_module_MOD_amrex_parallel_reduce_min_rs in AMReX_parallel_mod.o
      ___amrex_parallel_module_MOD_amrex_parallel_reduce_min_iv in AMReX_parallel_mod.o
      ___amrex_parallel_module_MOD_amrex_parallel_reduce_min_is in AMReX_parallel_mod.o
      ___amrex_parallel_module_MOD_amrex_parallel_reduce_max_rv in AMReX_parallel_mod.o
      ___amrex_parallel_module_MOD_amrex_parallel_reduce_max_rs in AMReX_parallel_mod.o
      ___amrex_parallel_module_MOD_amrex_parallel_reduce_max_iv in AMReX_parallel_mod.o
      ...
  "_mpi_comm_dup_", referenced from:
      ___amrex_parallel_module_MOD_amrex_parallel_init in AMReX_parallel_mod.o
  "_mpi_comm_free_", referenced from:
      ___amrex_parallel_module_MOD_amrex_parallel_finalize in AMReX_parallel_mod.o
  "_mpi_comm_rank_", referenced from:
      ___amrex_parallel_module_MOD_amrex_parallel_init in AMReX_parallel_mod.o
  "_mpi_comm_size_", referenced from:
      ___amrex_parallel_module_MOD_amrex_parallel_init in AMReX_parallel_mod.o
  "_mpi_finalize_", referenced from:
      _MAIN__ in main.o
      ___amrex_parallel_module_MOD_amrex_parallel_finalize in AMReX_parallel_mod.o
  "_mpi_init_", referenced from:
      _MAIN__ in main.o
      ___amrex_parallel_module_MOD_amrex_parallel_init in AMReX_parallel_mod.o
  "_mpi_initialized_", referenced from:
      ___amrex_parallel_module_MOD_amrex_parallel_init in AMReX_parallel_mod.o
  "_mpi_reduce_", referenced from:
      ___amrex_parallel_module_MOD_amrex_parallel_reduce_min_rv in AMReX_parallel_mod.o
      ___amrex_parallel_module_MOD_amrex_parallel_reduce_min_rs in AMReX_parallel_mod.o
      ___amrex_parallel_module_MOD_amrex_parallel_reduce_min_iv in AMReX_parallel_mod.o
      ___amrex_parallel_module_MOD_amrex_parallel_reduce_min_is in AMReX_parallel_mod.o
      ___amrex_parallel_module_MOD_amrex_parallel_reduce_max_rv in AMReX_parallel_mod.o
      ___amrex_parallel_module_MOD_amrex_parallel_reduce_max_rs in AMReX_parallel_mod.o
      ___amrex_parallel_module_MOD_amrex_parallel_reduce_max_iv in AMReX_parallel_mod.o
      ...
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
gmake: *** [../../..//Tools/GNUMake/Make.rules:28: main3d.gnu.DEBUG.MPI.ex] Error 1

**Comment by zingale** [comment]
This it's the Fortran part of the MPI library.  You may need to explicitly add the Fortran MPI library to your link line.

**Comment by WeiqunZhang** [comment]
What do you get from `make  USE_MPI=TRUE  print-mpif90_link_flags`?

**Comment by samcable** [comment]
Thanks, zingale.  That was the problem.  Just for the aid of the next slob, I compiled one of my own little test codes with gfortran and the "-v" flag, and got the path and directories that gfortran links in:

-L/usr/local/lib -lmpifort -lmpi -lpmpi -l gfortran -shared-libgcc

Then I created Make.local and put that in where it says you might have to supply the path to the Fortran MPI.  It then compiled without a problem.  Thanks!
